lib/Logger.ts

```diff
@@ -74,29 +90,29 @@ export default class Logger {
      * Caches an informational message locally, to be sent to the server if
      * needed later.
      *
-     * @param {String} message The message to log.
-     * @param {Boolean} sendNow if true, the message will be sent right away.
-     * @param {Object|String} parameters The parameters to send along with the message
-     * @param {Boolean} onlyFlushWithOthers A request will never be sent to the server if all loglines have this set to true
+     * @param message The message to log.
+     * @param sendNow if true, the message will be sent right away.
+     * @param parameters The parameters to send along with the message
+     * @param onlyFlushWithOthers A request will never be sent to the server if all loglines have this set to true
      */
-    info(message, sendNow = false, parameters = '', onlyFlushWithOthers = false) {
+    info(message: string, sendNow = false, parameters: Parameters = '', onlyFlushWithOthers = false) {
         const msg = `[info] ${message}`;
         this.add(msg, parameters, sendNow, onlyFlushWithOthers);
     }
 
     /**
      * Logs an alert.
      *
-     * @param {String} message The message to alert.
-     * @param {Object|String} parameters The parameters to send along with the message
-     * @param {Boolean} includeStackTrace Must be disabled for testing
+     * @param message The message to alert.
+     * @param parameters The parameters to send along with the message
+     * @param includeStackTrace Must be disabled for testing
      */
-    alert(message, parameters = {}, includeStackTrace = true) {
+    alert(message: string, parameters: Parameters = {}, includeStackTrace = true) {
         const msg = `[alrt] ${message}`;
         const params = parameters;
 
         if (includeStackTrace) {
-            params.stack = JSON.stringify(new Error().stack);
+            (params as Record<string, unknown>).stack = JSON.stringify(new Error().stack);
```

Comment:
Shouldn't we check if `params` is a object instead of doing the assertion? It would be safer

Reply:
Adjusted 👍